### PR TITLE
Update contributing.md

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Ref https://github.com/babel/babel/issues/16492#issuecomment-2111508764
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

In this PR we update the project's build requirements and promote the usage of `yarn jest`. Compared to `make test-only`, contributors can refer to Jest's docs for more cli options.

### CODE REVIEW

**Review by nicolo-ribaudo — APPROVED** [review]
Can we also completely remove `TEST_ONLY`/`TEST_GREP` from our scripts?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -113,49 +115,55 @@ $ make fix
...
+$ TEST_ONLY=babel-cli make test-only
 ```
 
 </details>
 <br>
 
 #### Run a subset of tests
 
-Use the `TEST_GREP` variable to run a subset of tests by name:
+Use the [Jest CLI `-t` option](https://jestjs.io/docs/cli#--testnamepatternregex) to to run a subset of tests with a name that matches the regex:
```
Comment (JLHwung) [inline CONTRIBUTING.md:163]:
```suggestion
Use the [Jest CLI `-t` option](https://jestjs.io/docs/cli#--testnamepatternregex) to run a subset of tests with a name that matches the regex:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -169,40 +177,53 @@ packages/babel-plugin-transform-arrow-functions/test/fixtures/arrow-functions/de
...
 <br>
 
-You can combine `TEST_DEBUG` with `TEST_GREP` or `TEST_ONLY` to debug a subset of tests. If you plan to stay long in the debugger (which you'll likely do!), you may increase the test timeout by editing [test/testSetupFile.js](https://github.com/babel/babel/blob/main/test/testSetupFile.js).
+You can combine `-i` with `-t` to debug a subset of tests. For example,
+
+```sh
+yarn run --inspect-brk jest -i babel-plugin-arrow-functions -t "destructuring parameters"
+```
+
+If you plan to stay long in the debugger (which you'll likely do!), you may increase the test timeout by editing [test/testSetupFile.js](https://github.com/babel/babel/blob/main/test/testSetupFile.js).
```
Comment (JLHwung) [inline CONTRIBUTING.md:215]: This file has been removed. To increase the timeout we can set the `--testTimeout` directly.

**Comment by JLHwung** [comment]
> Can we also completely remove TEST_ONLY/TEST_GREP from our scripts?

Personally I don't use them either. @liuxingbaoyu Do you think they should be kept?

**Comment by liuxingbaoyu** [comment]
Thanks for mentioning it! I didn't use them.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -169,40 +177,57 @@ packages/babel-plugin-transform-arrow-functions/test/fixtures/arrow-functions/de
...
+
+```sh
+yarn run --inspect-brk jest -i babel-plugin-arrow-functions -t "destructuring parameters"
+```
+
+If you plan to stay long in the debugger (which you'll likely do!), you may increase the test timeout by the [Jest CLI `--testTimeout` option](https://jestjs.io/docs/cli#--testtimeoutnumber). For example to increase test timeout to 500 seconds
+
+```sh
+yarn run --inspect-brk jest -i babel-parser -t "my new test" --testTimeout=500000
+```
```
Comment (liuxingbaoyu) [inline CONTRIBUTING.md:219]: Great example! Maybe `--runInBand` is useful here too.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -37,9 +37,9 @@ Feel free to check out the `#discussion`/`#development` channels on our [Slack](
 
 ## Developing
 
-_Node_: Check that Node is [installed](https://nodejs.org/en/download/) with version `^12.20 || >= 14.13`. You can check this with `node -v`.
+_Node_: Check that Node is [installed](https://nodejs.org/en/download/) with version `^20.10.0 || >=22.0.0`. You can check this with `node -v`.
 
-_Yarn_: Make sure that Yarn 1 is [installed](https://classic.yarnpkg.com/en/docs/install) with version >= `1.19.0`.
+_Yarn_: Make sure that Yarn is [installed](https://yarnpkg.com/getting-started/install) with version `>=4.0.0`.
```
Comment (liuxingbaoyu) [inline CONTRIBUTING.md:42]: There seems to be no actual installation of yarn v4 here, unfortunately the yarn official website only leaves the corepack tutorial.
Let's change the wording to something like configure to support yarn v4?

**Review by liuxingbaoyu — APPROVED** [review]
👍

File: CONTRIBUTING.md (hunk 3 above)
Comment (JLHwung) [reply CONTRIBUTING.md:219]: Well `-i` is actually an alias for `--runInBand`.

Comment (liuxingbaoyu) [reply CONTRIBUTING.md:219]: Sorry missed it. :)

File: CONTRIBUTING.md (hunk 4 above)
Comment (JLHwung) [reply CONTRIBUTING.md:42]: Well upon the first timing running `corepack enable yarn`, corepack will download the latest yarn binary and "install" it anyway. Contributors should refer to the official yarn docs for how to install yarn, hence the updated link. 

Therefore I think the current wording is fine, if we change it to something like "configure corepack to install yarn 4" but the yarn community one day decides to change their install recommendation, we will have to sync with them. For the time being, simply put "installed" seems sufficient.

**Pull request merged**